code/src/App.js

```diff
@@ -1,9 +1,84 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+
+import { HAPPY_THOUGHTS_URL, LIKE_THOUGHT_URL } from './reusable/urls'
+
+import SendThought from './components/SendThought'
+import ThoughtsList from './components/ThoughtsList'
+import ErrorPopup from './components/ErrorPopup'
 
 export const App = () => {
+  const [thoughtsList, setThoughtsList] = useState([])
+  const [newThought, setNewThought] = useState('')
+  const [error, setError] = useState(false)
+  
+  useEffect(() => {
+    fetchThoughtsList()
+  }, [])
+
+  //GET
+  const fetchThoughtsList = () => {
+    fetch(HAPPY_THOUGHTS_URL)
+      .then(response => response.json())
+      .then(thoughts => setThoughtsList(thoughts))
+      .catch(err => console.error(err))
+  }
+
+  const onNewThoughtChange = (e) => {
+    setNewThought(e.target.value)
+  }
+
+  //POST: 
+  const handleFormSubmit = (e) => {
+  e.preventDefault()
+
+  const options = {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json'
+      },
+      body: JSON.stringify({ message: newThought })
+  }
+
+  fetch(HAPPY_THOUGHTS_URL, options)
+    .then(response => {
+      if (!response.ok) {
+        setError(true)
+        throw new Error ('Ups, something went wrong') //prints error message in Console for the use case: unspecified backend error
+      } else { 
+        setNewThought('')
+        return response.json()
+      }
+    })
+    .then(recievedThought => setThoughtsList([recievedThought, ...thoughtsList]))
+    .catch(err => console.error(err))
+  } 
+
+  const handleHeartsIncrease = (id) => {
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }
+    fetch(LIKE_THOUGHT_URL(id), options)
+      .then(res => res.json())
+      .then(() => fetchThoughtsList())
+      .catch(err => console.error(err))
+  }
+
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+    <>
+      <SendThought 
+        newThought={newThought}
+        onNewThoughtChange={onNewThoughtChange}
+        handleFormSubmit={handleFormSubmit}
+      />
+      <ThoughtsList 
+        thoughtsList={thoughtsList}
+        handleHeartsIncrease={handleHeartsIncrease}  
+      />
+      {error && <ErrorPopup setError={setError}/>}
```

Comment:
I love this way of handling errors  - it is not only elegantly constructed but also cute. It makes me want to cause errors 🤯